### ISSUE TITLE
RVM demands gcc@7, then uninstalls it

### ISSUE THREAD

**Issue by jtippett** [rvm/rvm#4548]
## Description

During installation of ruby 2.6.0, the install process fails because gcc@7 is not present. After installing it, which took 47 minutes, rvm immediately and hilariously deletes it and the ruby installation proceeds as normal.

## Steps to reproduce

`rvm install 2.6.0`

## Expected behavior

I expected rvm to install 2.6.0 without issue.

## Actual behavior

The install process fails because gcc@7 is not installed. After gcc@7 is installed, the installer is satisfied, deletes gcc@7, and proceeds as expected:

```
$ rvm install 2.6.0
Searching for binary rubies, this might take some time.
No binary rubies available for: osx/10.14/x86_64/ruby-2.6.0.
Continuing with compilation. Please read 'rvm help mount' to get more information on binary rubies.
Checking requirements for osx.
Error: No such keg: /usr/local/Cellar/gcc@7
Requirements installation failed with status: 1

$ brew install gcc@7
Updating Homebrew...
Warning: Building gcc@7 from source:
  The bottle needs the Xcode CLT to be installed.
==> Downloading https://ftp.gnu.org/gnu/gcc/gcc-7.4.0/gcc-7.4.0.tar.xz
######################################################################## 100.0%
==> ../configure --build=x86_64-apple-darwin18.2.0 --prefix=/usr/local/Cellar/gcc@7/7.4.0 --libdir=/
==> make
==> make install
🍺  /usr/local/Cellar/gcc@7/7.4.0: 1,399 files, 248.4MB, built in 47 minutes 23 seconds

$ rvm install 2.6.0
Searching for binary rubies, this might take some time.
No binary rubies available for: osx/10.14/x86_64/ruby-2.6.0.
Continuing with compilation. Please read 'rvm help mount' to get more information on binary rubies.
Checking requirements for osx.
Uninstalling /usr/local/Cellar/gcc@7/7.4.0... (1,399 files, 248.4MB)
Updating certificates bundle '/usr/local/etc/openssl@1.1/cert.pem'
Requirements installation successful.
Warning: DYLD_LIBRARY_PATH environment variable is set, this might interact with the compilation and ruby.
Installing Ruby from source to: /Users/james/.rvm/rubies/ruby-2.6.0, this may take a while depending on your cpu(s)...
ruby-2.6.0 - #downloading ruby-2.6.0, this may take a while depending on your connection...
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 13.9M  100 13.9M    0     0   191k      0  0:01:14  0:01:14 --:--:--  197k
ruby-2.6.0 - #extracting ruby-2.6.0 to /Users/james/.rvm/src/ruby-2.6.0.....
ruby-2.6.0 - #configuring.......................................................................
ruby-2.6.0 - #post-configuration.
ruby-2.6.0 - #compiling.....................................................................
ruby-2.6.0 - #installing...........
ruby-2.6.0 - #making binaries executable..
Installed rubygems 3.0.1 is newer than 2.7.8 provided with installed ruby, skipping installation, use --force to force installation.
ruby-2.6.0 - #gemset created /Users/james/.rvm/gems/ruby-2.6.0@global
ruby-2.6.0 - #importing gemset /Users/james/.rvm/gemsets/global.gems...............................-
ruby-2.6.0 - #generating global wrappers.......
ruby-2.6.0 - #gemset created /Users/james/.rvm/gems/ruby-2.6.0
ruby-2.6.0 - #importing gemsetfile /Users/james/.rvm/gemsets/default.gems evaluated to empty gem list
ruby-2.6.0 - #generating default wrappers.......
ruby-2.6.0 - #adjusting #shebangs for (gem irb erb ri rdoc testrb rake).
Install of ruby-2.6.0 - #complete 
Ruby was built without documentation, to build it run: rvm docs generate-ri
```

## Environment info

```
ruby-2.6.0:

  system:
    uname:        "Darwin destiny.local 18.2.0 Darwin Kernel Version 18.2.0: Mon Nov 12 20:24:46 PST 2018; root:xnu-4903.231.4~2/RELEASE_X86_64 x86_64"
    name:         "OSX"
    version:      "10.14"
    architecture: "x86_64"
    bash:         "/bin/bash => GNU bash, version 3.2.57(1)-release (x86_64-apple-darwin18)"
    zsh:          "/bin/zsh => zsh 5.3 (x86_64-apple-darwin18.0)"
    remote_path:  "osx/10.14/x86_64"
    xcode:        "10.1"

  rvm:
    version:      "1.29.7 (master)"
    updated:      "9 hours 58 minutes 41 seconds ago"
    path:         "/Users/james/.rvm"
    autolibs:     "[4] Allow RVM to use package manager if found, install missing dependencies, install package manager (only OS X)."

  ruby:
    interpreter:  "ruby"
    version:      "2.6.0p0"
    date:         "2018-12-25"
    platform:     "x86_64-darwin18"
    patchlevel:   "2018-12-25 revision 66547"
    full_version: "ruby 2.6.0p0 (2018-12-25 revision 66547) [x86_64-darwin18]"

  homes:
    gem:          "/Users/james/.rvm/gems/ruby-2.6.0"
    ruby:         "/Users/james/.rvm/rubies/ruby-2.6.0"

  binaries:
    ruby:         "/Users/james/.rvm/rubies/ruby-2.6.0/bin/ruby"
    irb:          "/Users/james/.rvm/rubies/ruby-2.6.0/bin/irb"
    gem:          "/Users/james/.rvm/rubies/ruby-2.6.0/bin/gem"
    rake:         "/Users/james/.rvm/rubies/ruby-2.6.0/bin/rake"

  environment:
    PATH:         "/Users/james/.rvm/gems/ruby-2.6.0/bin:/Users/james/.rvm/gems/ruby-2.6.0@global/bin:/Users/james/.rvm/rubies/ruby-2.6.0/bin:/Users/james/.rvm/bin:/Users/james/.fastlane/bin:/Users/james/torch/install/bin:/Users/james/google-cloud-sdk/bin:/usr/local/sbin:/Users/james/.fastlane/bin:/Users/james/torch/install/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/usr/local/bin:/Users/james/Library/Android/sdk/tools:/Users/james/Library/Android/sdk/platform-tools:/Users/james/go/bin:/usr/local/opt/go/libexec/bin:/usr/local/bin:/Users/james/Library/Android/sdk/tools:/Users/james/Library/Android/sdk/platform-tools:/Users/james/go/bin:/usr/local/opt/go/libexec/bin"
    GEM_HOME:     "/Users/james/.rvm/gems/ruby-2.6.0"
    GEM_PATH:     "/Users/james/.rvm/gems/ruby-2.6.0:/Users/james/.rvm/gems/ruby-2.6.0@global"
    MY_RUBY_HOME: "/Users/james/.rvm/rubies/ruby-2.6.0"
    IRBRC:        "/Users/james/.rvm/rubies/ruby-2.6.0/.irbrc"
    RUBYOPT:      ""
    gemset:       ""

```

**Comment by FlowerWrong** [comment]
+1
After I install gcc@7 with `brew install gcc@7`.
Exec `rvm install ruby-2.5.3` will uninstall gcc@7. It is a for loop.

**Comment by crowell** [comment]
maybe this will fix it?

https://gist.github.com/crowell/b6a8f8cd33370a8808ca0f782d36467b

**Comment by pkuczynski** [comment]
Same issue as #4546. Should be fixed with #4557 and #4556. Please do `rvm get master` and try again. Let us know in case of the problem still existing.

**Comment by marcandre** [comment]
Still uninstalls gcc@7 for me

```
[22:47]$ rvm get master
Downloading https://get.rvm.io
No GPG software exists to validate rvm-installer, skipping.
Downloading https://github.com/rvm/rvm/archive/master.tar.gz
Upgrading the RVM installation in /Users/mal/.rvm/
    RVM PATH line found in /Users/mal/.mkshrc /Users/mal/.profile /Users/mal/.bashrc /Users/mal/.zshrc.
    RVM sourcing line found in /Users/mal/.profile /Users/mal/.bash_profile /Users/mal/.zlogin.
Upgrade of RVM in /Users/mal/.rvm/ is complete.
  * It looks like some old stuff is laying around RVM, you can cleanup with: rvm cleanup all

  * No new notes to display.

RVM reloaded!
[22:48]$ rvm install 2.6.0
ruby-2.6.0 - #removing src/ruby-2.6.0..
Searching for binary rubies, this might take some time.
No binary rubies available for: osx/10.14/x86_64/ruby-2.6.0.
Continuing with compilation. Please read 'rvm help mount' to get more information on binary rubies.
Checking requirements for osx.
Uninstalling gcc@7... (1,399 files, 256.2MB)
```

**Comment by pkuczynski** [comment]
See #4558

**Comment by pkuczynski** [comment]
We just released a new version 1.29.7 including this fix.

[Become a backer](https://opencollective.com/rvm#backer) and support us with a small donation to help us continue our activities. Thank you if you already one! 🙏
